libraries/botbuilder-dialogs-adaptive/src/recognizers/regexRecognizer.ts

```diff
@@ -71,13 +71,15 @@ export class RegexRecognizer implements Recognizer {
                     if (match.groups) {
                         for (const name in match.groups) {
                             const text = match.groups[name];
-                            const entity: Entity = {
-                                type: name,
-                            };
-                            entity['text'] = text;
-                            entity['start'] = match.index;
-                            entity['end'] = match.index + text.length;
-                            entityPool.push(entity);
+                            if (text) {
+                                const entity: Entity = {
+                                    type: name,
+                                };
+                                entity['text'] = text;
+                                entity['start'] = match.index;
+                                entity['end'] = match.index + text.length;
```

Comment:
This smells a little, but it's the same code as in C#. What's an example of the `"name"` that's used as the value for `Entity.type`?

We're using the Entity as defined in the [protocol spec](https://github.com/microsoft/botframework-obi/blob/master/protocols/botframework-activity/botframework-activity.md#Entity); but it's just holding numbers and strings here. 

The instancePool down below looks like it's expecting a [ModelResult](https://github.com/microsoft/botbuilder-js/blob/master/libraries/botbuilder-dialogs/src/choices/modelResult.ts) ([C#](https://github.com/microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder.Dialogs/Choices/ModelResult.cs)), not a low-level Entity. @stevenic @tomlm or @chrimc62 can you chime in?

Reply:
Yes, we have `new Entity(groupName)` in C#, I'm not quite sure about this, but we may need to update this if you found this is an incorrect implementation.

Reply:
Because this object eventually appears as `any` to the developer, it's fine to leave it as is... This is just a bit unfortunate 😞